### PR TITLE
Remove clientconfig and related global variables

### PR DESCRIPTION
**Pull request by josvazg**
**Description of the change**

Remove the `clientConfig` and namespace wrapper func `namespaceFromClientConfig` global variables.

**Benefits**

Disentangle the code so we can make further enhancements, such as get rid of the `run` method with a gazillion unnamed arguments.

### CODE REVIEW

**Comment by josvazg** [comment]
I made a mistake wen solving conflicts...

File: cmd/kubeseal/main.go [hunk 1]
```diff
@@ -108,6 +101,28 @@ func init() {
...
+func initUsualKubectlFlags(flagset *flag.FlagSet) *clientcmd.ConfigOverrides {
+	overrides := clientcmd.ConfigOverrides{}
+	kflags := clientcmd.RecommendedConfigOverrideFlags("")
+	clientcmd.BindOverrideFlags(&overrides, flagset, kflags)
+	return &overrides
+}
+
+func initClientFromFlags() clientcmd.ClientConfig {
+	return initClient(*kubeconfig, *globalOverrides, os.Stdout)
+}
```
Comment (alemorcuq) [inline cmd/kubeseal/main.go:113]: What are we expecting to read from `stdout`? I've noticed the original code reads from `stdin`, why this change?

File: cmd/kubeseal/main.go [hunk 2]
```diff
@@ -108,6 +101,28 @@ func init() {
...
+
+func initClient(kubeConfigPath string, cfgOverrides clientcmd.ConfigOverrides, w io.Reader) clientcmd.ClientConfig {
```
Comment (alemorcuq) [inline cmd/kubeseal/main.go:115]:
```suggestion
func initClient(kubeConfigPath string, cfgOverrides clientcmd.ConfigOverrides, r io.Reader) clientcmd.ClientConfig {
```

Afaik, it is a common practice to use `r` for Readers and `w` for Writers.

File: cmd/kubeseal/main.go [hunk 3]
```diff
@@ -71,17 +71,17 @@ var (
...
-	clientConfig clientcmd.ClientConfig
-
-	// testing hook for clientConfig.Namespace()
-	namespaceFromClientConfig = func() (string, bool, error) { return clientConfig.Namespace() }
 )
 
-func init() {
+type namespaceFn func() (string, bool, error)
+
+func initFlags() {
```
Comment (alemorcuq) [inline cmd/kubeseal/main.go:84]: Why are we replacing `init` with `initFlags`?

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (josvazg) [reply cmd/kubeseal/main.go:113]: This is a mistake. Will fix it

File: cmd/kubeseal/main.go (hunk 3 above)
Comment (josvazg) [reply cmd/kubeseal/main.go:84]: `init()` is magic, it is always executed before the `main` function.

It means that such initialization is happening no matter what, and in this case that is NOT required. You can make the case for when you are, for instance, loading device or SQL driver implementations coming from dependencies.

Using `initFlags` instead we have full control on when the flags are parsed or not.

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (josvazg) [inline cmd/kubeseal/main.go:113]:
```suggestion
func initClientFromFlags() clientcmd.ClientConfig {
	return initClient(*kubeconfig, *globalOverrides, os.Stdin)
}
```

File: cmd/kubeseal/main.go (hunk 2 above)
Comment (josvazg) [reply cmd/kubeseal/main.go:115]: Right! Thanks!

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (josvazg) [reply cmd/kubeseal/main.go]: Done

File: cmd/kubeseal/main_test.go [hunk 4]
```diff
@@ -102,6 +93,21 @@ func tmpfile(t *testing.T, contents []byte) string {
...
+	err := flagset.Parse([]string{"-n", "default"})
+	if err != nil {
+		fmt.Printf("flagset parse err: %v\n", err)
+		os.Exit(1)
+	}
+	return testOverrides
+}
+
+func testClientConfig() clientcmd.ClientConfig {
+	return initClient("", *testConfigOverrides(), os.Stdout)
```
Comment (josvazg) [inline cmd/kubeseal/main_test.go:108]:
```suggestion
	return initClient("", *testConfigOverrides(), os.Stdin)
```

File: cmd/kubeseal/main.go [hunk 5]
```diff
@@ -108,6 +101,28 @@ func init() {
...
+
+func initClientFromFlags() clientcmd.ClientConfig {
+	return initClient(*kubeconfig, *globalOverrides, os.Stdin)
+}
+
+func initClient(kubeConfigPath string, cfgOverrides clientcmd.ConfigOverrides, r io.Reader) clientcmd.ClientConfig {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+	loadingRules.ExplicitPath = kubeConfigPath
+	return clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, &cfgOverrides, w)
```
Comment (alemorcuq) [inline cmd/kubeseal/main.go:119]:
```suggestion
	return clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, &cfgOverrides, r)
```

**Review by alemorcuq — CHANGES_REQUESTED** [review]

**Review by alemorcuq — APPROVED** [review]
